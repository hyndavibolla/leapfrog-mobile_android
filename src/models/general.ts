import { Dispatch } from 'react';
import Auth0 from 'react-native-auth0';

import { ApiService } from '_services/ApiService';
import { AppService } from '_services/AppService';
import { CCPAService } from '_services/CCPAService';
import { ICrashlyticsService } from '_services/CrashlyticsService';
import { EventTrackerService } from '_services/EventTrackerService';
import { IHttpService } from '_services/HttpService';
import { NativeHelperService } from '_services/NativeHelperService';
import { Logger } from '_services/Logger';
import { IPerformanceTrackingService } from '_services/PerformanceTrackingService';
import { RaiseService } from '_services/RaiseService';
import { RemoteConfigService } from '_services/RemoteConfigService';
import { ICardLinkState } from '_state_mgmt/cardLink/state';
import { ICoreState } from '_state_mgmt/core/state';
import { IGameState } from '_state_mgmt/game/state';
import { IGiftCardState } from '_state_mgmt/giftCard/state';
import { IMessagesState } from '_state_mgmt/messages/state';
import { IMissionState } from '_state_mgmt/mission/state';
import { IRewardState } from '_state_mgmt/reward/state';
import { ISurveyState } from '_state_mgmt/survey/state';
import { IUserState } from '_state_mgmt/user/state';

export interface IAction<T = string, P = any> {
  type: T;
  payload: P;
}

export type IReducer<S = any, A = any> = (state: S, action: IAction<A>) => S;

export interface Deps {
  apiService: ApiService;
  appService: AppService;
  crashlyticsService: ICrashlyticsService;
  httpService: IHttpService;
  performanceTrackingService: IPerformanceTrackingService;
  raiseService: RaiseService;
  nativeHelperService: NativeHelperService;
  stateSnapshot: { get: () => IGlobalState; set: (state: IGlobalState) => void };
  logger: Logger;
  eventTrackerService: EventTrackerService;
  ccpaService: CCPAService;
  remoteConfigService: RemoteConfigService;
  authService: Auth0;
}
export type IGlobalState = {
  core: ICoreState;
  user: IUserState;
  game: IGameState;
  mission: IMissionState;
  reward: IRewardState;
  survey: ISurveyState;
  messages: IMessagesState;
  cardLink: ICardLinkState;
  giftCard: IGiftCardState;
};

export interface IGlobalContext {
  state: IGlobalState;
  dispatch: Dispatch<IAction<any, any>>;
  deps: Deps;
}

export enum HTTP_METHOD {
  GET = 'GET',
  POST = 'POST',
  PUT = 'PUT',
  PATCH = 'PATCH',
  DELETE = 'DELETE'
}

export interface ITokenSet {
  accessToken: string;
  refreshToken: string;
  accessTokenExpiryTime: string;
  sywToken?: string;
  /**
   * @deprecated deviceId does not belong to ITokenSet anymore.
   */
  deviceId?: string;
  scope?: string;
  expiresIn?: number;
  tokenType?: string;
}

export enum FeatureFlag {
  API_OVERRIDE = 'API_OVERRIDE',
  TEST_ONLY = 'TEST_ONLY',
  ENV_BANNER = 'ENV_BANNER',
  BUTTON_DEBUG = 'BUTTON_DEBUG',
  LOG_LEVEL_INFO = 'LOG_LEVEL_INFO',
  LOG_LEVEL_DEBUG = 'LOG_LEVEL_DEBUG',
  LOG_LEVEL_WARN = 'LOG_LEVEL_WARN',
  LOG_LEVEL_ERROR = 'LOG_LEVEL_ERROR',
  LOG_VIEW = 'LOG_VIEW',
  STREAK_REDEEM_CARD = 'STREAK_REDEEM_CARD',
  SAILTHRU_TRACKING = 'SAILTHRU_TRACKING',
  SURVEY = 'SURVEY',
  PREDICTIVE_MAP_SEARCH = 'PREDICTIVE_MAP_SEARCH',
  MISSIONS_SEARCH = 'MISSIONS_SEARCH',
  SEE_ALL_TRANSACTIONS = 'SEE_ALL_TRANSACTIONS',
  WALLET_YOUR_GC = 'WALLET_YOUR_GC'
}

export type DateLike = Date | string | number;

export type HTMLString = string;

export interface IGameLevelUI {
  updatedAt: DateLike;
  level: number;
}

export interface ICCPASetting {
  allow: boolean;
}

export interface IOnboarding {
  seenOnboarding: boolean;
}

export enum ButtonCreativeType {
  HERO = 'hero',
  CAROUSEL = 'carousel',
  LIST = 'list',
  GRID = 'grid',
  DETAIL = 'detail',
  OTHER = 'other'
}

export interface ISearchHistoryItem {
  keyword: string;
  lastUpdatedAt: DateLike;
}

export enum TooltipKey {
  EARN = 'earn',
  REWARDS = 'rewards',
  MISSIONS = 'missions',
  STREAK = 'streak'
}

export interface IApiOverrideRule {
  key: string;
  description?: string;
  matcher: string; // reg exp
  status: number;
  response: any;
}

export namespace RemoteConfigModel {
  export interface IAppVersion {
    suggested: number;
    required: number;
  }

  export interface IEarnSection {
    name: string;
    order: number;
    visible: boolean;
    title: string;
    description: string;
    seeAllButton: boolean;
  }
}

export enum RUNTIME_ENV {
  LIVE = 'live',
  DEV_TEST = 'test'
}

export interface IMeasure {
  x: number;
  y: number;
  width: number;
  height: number;
}

export interface IExpirePointsBannerDataSet {
  lastPointsDate: DateLike;
  lastAcceptedDate: DateLike;
}

export interface ISailthruMessage {
  createdAt: string;
  custom: {
    [k: string]: string;
  };
  htmlText: string;
  id: string;
  isRead: boolean;
  text: string;
  title: string;
  type: string;
  cardImageUrl?: string;
  share: boolean;
}

export interface ITutorialStatus {
  isBannerWatched: boolean;
  date: string;
}
