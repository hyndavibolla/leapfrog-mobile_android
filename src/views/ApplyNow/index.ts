export { default as ApplyNow } from './ApplyNow';
