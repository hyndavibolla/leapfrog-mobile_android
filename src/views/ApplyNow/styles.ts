import { css } from '@emotion/native';

import { COLOR, FONT_FAMILY, FONT_SIZE, MEASURE } from '../../constants';

export const styles = {
  contentContainer: css`
    flex: 1;
    justify-content: center;
    background-color: ${COLOR.LIGHT_GRAY};
  `,
  switchContainer: css`
    margin: 36px 40px 24px 40px;
  `,
  imageContainer: css`
    margin: 0px 28px 16px 28px;
  `,
  image: css`
    align-self: center;
  `,
  sectionContainer: css`
    margin: 0 16px;
  `,
  title: css`
    font-family: ${FONT_FAMILY.HEAVY};
    text-align: center;
    letter-spacing: 0;
  `,
  ecmTitle: css`
    margin-horizontal: 52px;
    margin-bottom: 4px;
  `,
  nonEcmTitle: css`
    margin: 28px 28px 4px;
  `,
  nonEcmParagraph: css`
    margin-top: 8px;
    margin-bottom: 4px;
  `,
  text: css`
    text-align: center;
    font-size: ${FONT_SIZE.SMALLER};
    line-height: ${FONT_SIZE.HUGE};
    color: ${COLOR.DARK_GRAY};
  `,
  link: css`
    color: ${COLOR.PRIMARY_BLUE};
    letter-spacing: 0;
  `,
  linkNonECM: css`
    margin-bottom: 8px;
    margin-top: 0px;
  `,
  buttonContainer: css`
    margin: 8px 40px 20px;
  `,
  buttonOuter: css`
    width: 100%;
  `,
  buttonInner: css`
    padding: 8px 0;
    background-color: ${COLOR.PRIMARY_BLUE};
  `,
  buttonText: css`
    font-size: ${FONT_SIZE.SMALL};
    letter-spacing: 0;
    line-height: ${FONT_SIZE.HUGE};
    font-family: ${FONT_FAMILY.BOLD};
  `,
  modalContainer: css`
    flex: 1;
    align-items: center;
    background-color: ${COLOR.WHITE};
    padding-top: ${MEASURE.MODAL_PADDING}px;
    border-radius: 10px;
  `,
  modalTitle: css`
    padding-top: 40px;
  `,
  modalImageContainer: css`
    margin: 20px 28px 16px;
  `,
  modalSectionContainer: css`
    margin: 0 20px 20px;
  `,
  modalText: css`
    font-size: ${FONT_SIZE.SMALLER};
    letter-spacing: 0;
    text-align: center;
  `,
  modalImage: css`
    align-self: center;
    width: 80px;
    height: 50px;
    border-radius: 4px;
  `,
  modalBenefitsContainer: css`
    background-color: ${COLOR.LIGHT_GRAY};
    width: 100%;
  `,
  modalBenefits: css`
    margin-top: 20px;
    align-self: center;
  `,
  cardLinkContainer: css`
    flex: 1;
    align-items: center;
    background-color: ${COLOR.LIGHT_GRAY};
    width: 100%;
    padding-bottom: 52px;
    padding-top: 8px;
  `,
  cardLink: css`
    color: ${COLOR.BLACK};
    font-size: ${FONT_SIZE.EXTRA_TINY};
    text-decoration: underline;
  `,
  modal: css`
    padding-bottom: 0px;
    padding-top: 8px;
    height: 95%;
  `,
  largeContainer: css`
    flex: 1;
    align-items: center;
    background-color: ${COLOR.LIGHT_GRAY};
    padding-top: 20px;
  `,
  closeButton: css`
    top: 20px;
  `
};
