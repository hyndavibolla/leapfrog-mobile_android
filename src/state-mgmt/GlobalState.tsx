import React, { createContext, memo, PropsWithChildren, useReducer, Dispatch, useEffect } from 'react';

import { IGlobalState, Deps, IReducer, IGlobalContext } from '_models/general';
import { reducer as coreReducer } from './core/reducer';
import { reducer as userReducer } from './user/reducer';
import { reducer as gameReducer } from './game/reducer';
import { reducer as missionReducer } from './mission/reducer';
import { reducer as rewardReducer } from './reward/reducer';
import { reducer as surveyReducer } from './survey/reducer';
import { reducer as messagesReducer } from './messages/reducer';
import { reducer as cardLinkReducer } from './cardLink/reducer';
import { reducer as giftCardReducer } from './giftCard/reducer';
import { initialState as coreInitialState } from './core/state';
import { initialState as userInitialState } from './user/state';
import { initialState as gameInitialState } from './game/state';
import { initialState as missionInitialState } from './mission/state';
import { initialState as rewardInitialState } from './reward/state';
import { initialState as surveyInitialState } from './survey/state';
import { initialState as messagesInitialState } from './messages/state';
import { initialState as cardLinkInitialState } from './cardLink/state';
import { initialState as giftCardInitialState } from './giftCard/state';
import { combineReducers } from '_utils/combineReducers';
import { GeneralModel } from '_models';
import { deepClone } from '_utils/deepClone';

export const getInitialState = () =>
  deepClone<IGlobalState>({
    core: coreInitialState,
    user: userInitialState,
    game: gameInitialState,
    mission: missionInitialState,
    reward: rewardInitialState,
    survey: surveyInitialState,
    messages: messagesInitialState,
    cardLink: cardLinkInitialState,
    giftCard: giftCardInitialState
  });

export const combinedReducer = combineReducers<GeneralModel.IGlobalState>({
  core: coreReducer,
  user: userReducer,
  game: gameReducer,
  mission: missionReducer,
  reward: rewardReducer,
  survey: surveyReducer,
  messages: messagesReducer,
  cardLink: cardLinkReducer,
  giftCard: giftCardReducer
});

export const GlobalContext = createContext<{ state: IGlobalState; dispatch: Dispatch<any>; deps: Deps }>({
  state: getInitialState(),
  dispatch: /* istanbul ignore next */ () => null,
  deps: null
} as IGlobalContext);

export const GlobalProvider = memo(
  ({
    children,
    deps,
    initState,
    combinedReducers
  }: PropsWithChildren<{
    deps: Deps;
    initState: IGlobalState;
    combinedReducers: IReducer<IGlobalState>;
  }>) => {
    const [state, dispatch] = useReducer(combinedReducers, initState);
    useEffect(() => deps.stateSnapshot.set(state), [state]); // eslint-disable-line react-hooks/exhaustive-deps
    return <GlobalContext.Provider value={{ state, dispatch, deps } as IGlobalContext}>{children}</GlobalContext.Provider>;
  }
);
