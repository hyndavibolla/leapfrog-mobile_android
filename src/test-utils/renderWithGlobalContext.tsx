import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
