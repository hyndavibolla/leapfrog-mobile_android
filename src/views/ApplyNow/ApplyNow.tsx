import React, { memo, useState, useContext, useCallback, useMemo } from 'react';
import { View, Image, ScrollView } from 'react-native';
import { StackNavigationProp } from '@react-navigation/stack';

import { Modal, ModalSize } from '_components/Modal';
import { Title, TitleType } from '_components/Title';
import { Text } from '_components/Text';
import { Button } from '_components/Button';

import { ENV, FONT_FAMILY, ForterActionType, ROUTES, TealiumEventType } from '_constants';

import { useEventTracker } from '_state_mgmt/core/hooks';
import { GlobalContext } from '_state_mgmt/GlobalState';
import { useTestingHelper } from '_utils/useTestingHelper';

import CreditCardBenefits from '_assets/credit-card/benefits@3x.svg';

import { styles } from './styles';

export interface Props {
  navigation: StackNavigationProp<any>;
}

export const ApplyNow = ({ navigation }: Props) => {
  const [isDetailsModalOpen, setIsDetailsModalOpen] = useState<boolean>(false);
  const { deps } = useContext(GlobalContext);
  const { getTestIdProps } = useTestingHelper('apply-now');
  const ecmWidth = useMemo(() => deps.nativeHelperService.dimensions.getWindowWidth() - 60, [deps.nativeHelperService.dimensions]);
  const ecmHeight = useMemo(() => ecmWidth / 1.4, [ecmWidth]);
  const { trackUserEvent } = useEventTracker();
  const isIOS = useMemo(() => deps.nativeHelperService.platform.OS === 'ios', [deps.nativeHelperService.platform]);
  const benefitsImageScale = useMemo(() => (isIOS ? 1.6 : 1.2), [isIOS]);

  const onApplyingForCardPress = useCallback(() => {
    trackUserEvent(TealiumEventType.CARD_APPLICATION, { event_type: TealiumEventType.INITIATE_CITI_APPLICATION }, ForterActionType.TAP);
    navigation.navigate(ROUTES.FUSION_VIEWER, { routeToReturn: ROUTES.MAIN_TAB.WALLET });
  }, [navigation, trackUserEvent]);

  return (
    <ScrollView contentContainerStyle={styles.contentContainer} {...getTestIdProps('non-ecm-content')}>
      <View style={styles.imageContainer}>
        <Image source={require('_assets/credit-card/creditCard.png')} style={styles.image as any} />
      </View>
      <View style={styles.sectionContainer}>
        <Title type={TitleType.HEADER} style={[styles.title, styles.nonEcmTitle]}>
          Get rewarded on everyday purchases.
        </Title>
        <Text style={[styles.text, styles.nonEcmParagraph]}>
          Start earning Shop Your Way速 Points {'\n'} on eligible everyday purchases {'\n'} with a Shop Your Way Mastercard速
        </Text>
        <Text
          style={[styles.text, styles.nonEcmParagraph, styles.link, styles.linkNonECM]}
          onPress={() => setIsDetailsModalOpen(true)}
          {...getTestIdProps('see-details-link')}
        >
          See details
        </Text>
        <View style={styles.buttonContainer}>
          <Button
            onPress={onApplyingForCardPress}
            innerContainerStyle={[styles.buttonOuter, styles.buttonInner]}
            textStyle={styles.buttonText}
            {...getTestIdProps('apply-btn')}
          >
            Apply Now
          </Button>
        </View>
      </View>
      <Modal
        visible={isDetailsModalOpen}
        size={ModalSize.FULL_SCREEN}
        onClose={() => setIsDetailsModalOpen(false)}
        onPressOutside={() => setIsDetailsModalOpen(false)}
        showCloseButton
        style={styles.modal}
        closeButtonStyle={styles.closeButton}
      >
        <View style={styles.modalContainer} {...getTestIdProps('details-modal-content')}>
          <Text font={FONT_FAMILY.BOLD} style={styles.modalTitle}>
            SHOP YOUR WAY MASTERCARD速
          </Text>
          <View style={styles.modalImageContainer}>
            <Image source={require('_assets/credit-card/creditCard.png')} style={styles.modalImage as any} />
          </View>
          <View style={styles.modalSectionContainer}>
            <Text font={FONT_FAMILY.BOLD} style={styles.modalText}>
              The Shop Your Way Mastercard速 rewards members with more ways to earn points on eligible everyday purchases.
            </Text>
          </View>
          <View style={styles.largeContainer}>
            <View style={styles.modalBenefitsContainer}>
              <CreditCardBenefits style={styles.modalBenefits} width={ecmWidth * benefitsImageScale} height={ecmHeight * benefitsImageScale} />
            </View>
            <View style={styles.cardLinkContainer}>
              <Text style={styles.cardLink} onPress={() => deps.nativeHelperService.linking.openURL(ENV.FUSION.APPLY_NOW_URI)} {...getTestIdProps('card-link')}>
                *Shop Your Way additional category earn program
              </Text>
            </View>
          </View>
        </View>
      </Modal>
    </ScrollView>
  );
};

export default memo(ApplyNow);
