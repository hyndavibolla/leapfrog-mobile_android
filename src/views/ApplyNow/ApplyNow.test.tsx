import React from 'react';
import { fireEvent } from '@testing-library/react-native';
import { act } from 'react-test-renderer';

import { Deps, IGlobalState } from '_models/general';
import { getMockDeps } from '_test_utils/getMockDeps';
import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';
import { ApplyNow, Props } from './ApplyNow';
import { wait } from '_utils/wait';
import { ROUTES, ENV } from '_constants';
import { getInitialState } from '_state_mgmt/GlobalState';

describe('Apply Now', () => {
  let props: Props;
  let deps: Deps;
  let initialState: IGlobalState;

  beforeEach(() => {
    deps = getMockDeps();
    initialState = getInitialState();
    props = {
      navigation: { navigate: jest.fn() } as any
    };
  });

  it('should render', async () => {
    deps.nativeHelperService.platform.OS = 'ios';
    const { toJSON, queryByTestId } = renderWithGlobalContext(<ApplyNow {...props} />, deps);
    await act(() => wait(0));
    expect(toJSON()).toMatchSnapshot();
    expect(queryByTestId('apply-now-non-ecm-content')).toBeTruthy();
  });

  it('should show details modal', async () => {
    const { getByTestId, queryByTestId } = renderWithGlobalContext(<ApplyNow {...props} />);
    await act(() => wait(0));
    fireEvent.press(getByTestId('apply-now-see-details-link'));
    expect(queryByTestId('apply-now-details-modal-content')).toBeTruthy();
    fireEvent.press(getByTestId('modal-backdrop'));
  });

  it('should navigate to fusion webview', async () => {
    const route = { params: { routeToReturn: ROUTES.MAIN_TAB.WALLET } };
    const { getByTestId } = renderWithGlobalContext(<ApplyNow {...props} />);
    await act(() => wait(0));
    fireEvent.press(getByTestId('apply-now-apply-btn'));
    expect(props.navigation.navigate).toBeCalledWith(expect.any(String), route.params);
  });

  it('should navigate to SYW web', async () => {
    initialState.game.current.memberships.userHasSywCard = false;
    const { getByTestId } = renderWithGlobalContext(<ApplyNow {...props} />, deps, initialState);
    await act(() => wait(0));
    fireEvent.press(getByTestId('apply-now-see-details-link'));
    fireEvent.press(getByTestId('apply-now-card-link'));
    expect(deps.nativeHelperService.linking.openURL).toBeCalledWith(ENV.FUSION.APPLY_NOW_URI);
  });
});
