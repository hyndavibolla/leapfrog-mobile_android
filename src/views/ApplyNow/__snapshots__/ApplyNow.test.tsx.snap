// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apply Now should render 1`] = `
<RCTScrollView
  accessibilityLabel="apply-now-non-ecm-content"
  accessible={true}
  contentContainerStyle={
    Object {
      "backgroundColor": "#F5F7FA",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
    }
  }
  testID="apply-now-non-ecm-content"
>
  <View>
    <View
      style={
        Object {
          "marginBottom": 16,
          "marginLeft": 28,
          "marginRight": 28,
          "marginTop": 0,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/credit-card/creditCard.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "marginBottom": 0,
          "marginLeft": 16,
          "marginRight": 16,
          "marginTop": 0,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#262C2F",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 20,
              "lineHeight": 30,
            },
            Array [
              Object {
                "fontFamily": "SFProDisplay-Heavy",
                "letterSpacing": 0,
                "textAlign": "center",
              },
              Object {
                "marginBottom": 4,
                "marginLeft": 28,
                "marginRight": 28,
                "marginTop": 28,
              },
            ],
          ]
        }
        type="titleHeader"
      >
        Get rewarded on everyday purchases.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Medium",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Array [
              Object {
                "color": "#909BA0",
                "fontSize": 16,
                "lineHeight": 26,
                "textAlign": "center",
              },
              Object {
                "marginBottom": 4,
                "marginTop": 8,
              },
            ],
          ]
        }
      >
        Start earning Shop Your Way® Points 
        

         on eligible everyday purchases 
        

         with a Shop Your Way Mastercard®
      </Text>
      <Text
        accessibilityLabel="apply-now-see-details-link"
        accessible={true}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Medium",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Array [
              Object {
                "color": "#909BA0",
                "fontSize": 16,
                "lineHeight": 26,
                "textAlign": "center",
              },
              Object {
                "marginBottom": 4,
                "marginTop": 8,
              },
              Object {
                "color": "#0066CC",
                "letterSpacing": 0,
              },
              Object {
                "marginBottom": 8,
                "marginTop": 0,
              },
            ],
          ]
        }
        testID="apply-now-see-details-link"
      >
        See details
      </Text>
      <View
        style={
          Object {
            "marginBottom": 20,
            "marginLeft": 40,
            "marginRight": 40,
            "marginTop": 8,
          }
        }
      >
        <View
          accessibilityLabel="apply-now-apply-btn"
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="apply-now-apply-btn"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0066CC",
                  "borderBottomLeftRadius": 30,
                  "borderBottomRightRadius": 30,
                  "borderTopLeftRadius": 30,
                  "borderTopRightRadius": 30,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 8,
                },
                Array [
                  Object {
                    "width": "100%",
                  },
                  Object {
                    "backgroundColor": "#0066CC",
                    "paddingBottom": 8,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 8,
                  },
                ],
              ]
            }
          >
            <View
              style={false}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#909BA0",
                      "fontFamily": "SFProDisplay-Medium",
                      "fontSize": 16,
                      "lineHeight": 22,
                    },
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 14,
                        "textAlign": "center",
                      },
                      Object {
                        "fontFamily": "SFProDisplay-Bold",
                        "fontSize": 18,
                        "letterSpacing": 0,
                        "lineHeight": 26,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Apply Now
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <Modal
      accessibilityLabel="modal-container"
      accessible={true}
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      statusBarTranslucent={true}
      testID="modal-container"
      transparent={true}
      visible={false}
    />
  </View>
</RCTScrollView>
`;
